src/potato_udp.erl

```diff
@@ -16,62 +16,91 @@
 %% maps node_id to Pid of the process
 -type node_map() :: map().
 
--type state() :: {gen_udp:socket(), node_map()}.
+-type event_fn() :: fun( (atom(), any()) -> any() ).
 
+-record(udp_state, 
+	{
+	  %% my_registered_name :: any(),
+	  socket :: gen_udp:socket(),
+	  node_map :: node_map(),
+	  event_fn :: event_fn()
+	}).
 
-start_link(Port) ->
-    gen_server:start_link({local, potato_udp}, potato_udp, Port, []).
 
--spec init(udp_port()) -> {ok, state()}.
+-type state() :: #udp_state{}.
```

Comment:
:oo

Reply:
was too lazy to change state() to #udp_state{}

Reply:
such abstraction